[18.0][FIX] account_financial_report: replace old field company_id in account.account by company_ids

bguillot opened the pull request:
In v18 account.account field company_id as been replaced by company_ids

bguillot committed:
[FIX] account_financial_report: replace old field company_id in account.account by company_ids

florian-dacosta commented on account_financial_report/wizard/general_ledger_wizard.py:
```diff
@@ -108,7 +108,7 @@ def on_change_account_range(self):
             if self.company_id:
                 self.account_ids = self.account_ids.filtered(
-                    lambda a: a.company_id == self.company_id
+                    lambda a: self.company_id in a.company_ids
```
I think we could include company_ids in the search (with a if to increment the domain) and this way get rid of the filtered.
But it is more a refactore than a fix, and I believe we should do a large refactore of this module to be able to get a report for multiple companies (but necessarly all). It could be done at this time.

florian-dacosta reviewed (approved):
LGTM code review

alexis-via committed:
[FIX] Account filter on general ledger wizard

This commit both improves perfs by doing a single search() instead of
search() + filtered and also make account selection by interval really
work.

alexis-via commented:
fixes https://github.com/OCA/account-financial-reporting/issues/1335

alexis-via commented:
/ocabot merge minor
